### DISCUSSION TITLE
Help wanted?

### DISCUSSION THREAD

**Discussion by rjokelai** [sqids/sqids-javascript#2] [General]
Hi @4kimov, I would be interested in helping the JS version come alive. Anything I could do?

**Comment by 4kimov** [comment]
@rjokelai Thanks for pinging. Sure, there's plenty you can do -- the rough code is there, what's missing is proper linting, formatting, making all the scripts work properly, optimizations, packaging it properly, etc. Gave you access if you'd like to give it a go.

**Reply by rjokelai** [reply]
Thank you. Do you have any guidelines that you want to enforce across the implementations or am I free to do as I please here?

**Reply by 4kimov** [reply]
The major things are to have identical tests across implementations (which is already done for the JS version). I'd recommend leaving the README as is for now (with the exception of removing the "work-in-progress" notice), and focusing on just making all the npm run scripts work + making Github Actions pass.

**Comment by jcbbb** [comment]
Hello, I was just trying out the package, and it does not support `BigInt`. What would be a good way to support this or there any plan to? It basically requires everything that returns `Number` be casted to `BigInt` for calculation parts. What do you think?

**Reply by 4kimov** [reply]
Hello, thanks for the question. Let's move the discussion here: https://github.com/sqids/sqids-javascript/issues/4. No immediate plans to address this yet, but if you'd like to create a PR for it, I could review.

**Comment by mgred** [comment]
@4kimov what is the current state? Do you see anything (with priority) that needs to be done?

**Reply by 4kimov** [reply]
There's nothing that's urgent on the JS package. It's missing bigint support, but other than that fully usable.
